### ISSUE TITLE
Incorrect "no-useless-escape" warning

### ISSUE THREAD

**Issue by nbrady-techempower** [eslint/eslint#12209]
eslint version: 5.12.0
node version: 12.9.1
npm version: 6.10.2

I am getting a `no-useless-escape` warning in the following RegEx:

```javascript
str.replace(/\[(image|img)=(https?:\/\/[\d\w?=.&/:\-]+)\]/g, `<img src="$2" alt="$2">`;
```

`Line 51:  Unnecessary escape character: \-  no-useless-escape`

An escape is necessary for `-` in Character Sets when it's not denoting a range.

**Comment by nbrady-techempower** [comment]
I'm a liar. Once upon a time I believed you did need to escape hyphens in character sets. Great work team.

**Comment by platinumazure** [comment]
Hi @nbrady-techempower, hyphens don't need to be escaped in character classes if they are the first or last character in the character class. In all other locations in a character class, they must be escaped. Hope this helps!

**Comment by nbrady-techempower** [comment]
@platinumazure Thanks! I figured it out a few minutes too late.
